enhancements/authentication/pod-security-admission-autolabeling.md

```diff
@@ -0,0 +1,304 @@
+without breaking pod controller workloads
+2. Privileged users can opt-in/opt-out a namespace to and from the Pod Security admission
+
+### Non-Goals
+
+1. Make sure that bare pods (pods w/o pod controllers) continue working
+    - Both the admission systems are triggered on pod creation, possibly preventing the
+      API server from persisting the pod on admission denial. It is therefore impossible
+      to take the Pod as an input without touching code of both the admission controllers.
+      Such a code change would also very likely end up being rather unreasonable.
+2. Add the ability to tune the cluster-wide PSa configuration
+
+## Proposal
+
+### User Stories
+
+1. I would like my workloads to be admitted based on the current pod security standards
+   with minimal effort or no effort at all
+
+### Pod Security Admission Label Synchronization
+
+This section explains how pod security admission works and later elaborates on how to use
+the pod security labeling in order to synchronize SCC permissions into pod security admission
+restriction levels.
+
+The SCC permissions to PSa levels transformation gets explained at the end of this section.
+
+#### Pod Security Admission Introduction
+
+Pod Security admission validates pods' security according to the upstream
+[pod security standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/)
+and distinguishes three different security levels:
+- privileged - most privileged mode, anything is allowed
+- baseline   -  minimally restrictive policy which prevents known privilege escalations
+- restricted - heavily restricted policy, following current Pod hardening best practices
+
+By default, there is a cluster-global configuration which enforces the configured policies
+on pods and known workloads.
+
+It is possible to override the cluster-global policy enforce configuration on a per-namespace
+basis by using the `pod-security.kubernetes.io/enforce` label on given namespaces. It is also
+possible to exempt certain users, namespace and runtime classes from the admission completely.
+
+#### PSa Label Synchronization Controller
+
+This section describes a controller that is going to be a part of the cluster-policy-controller.
+
+In order to keep SCCs working alongside restricted cluster-wide PSa level, there must be
+a mechanism that would synchronize the PSa policy level for a namespace that workload runs
+in, so that the workload is allowed by both SCC and PSa admissions. For that purpose, a PSa
+Label Synchronization Controller (hereinafter the Controller) shall be built.
+
+The Controller watches changes of:
+- **namespaces**
+  - manual changes to the interesting labels of the namespaces that were
+    opted-in to label synchronization should be reverted
+- **roles** and **rolebindings** (and their **cluster variants**)
+  - to be able to properly assess SA capabilities in controlled namespaces
+- **SCCs**
```

Comment:
will users and groups in SCCs be honored or will you only honor RBAC based mappings?

Reply:
users and groups in SCCs need to be honored, too